src/Prometheus.AspNet/Classes/PrometheusHttpRequestModule.cs

```diff
@@ -9,24 +9,83 @@ namespace Prometheus.AspNet
     /// </summary>
     public class PrometheusHttpRequestModule : IHttpModule
     {
-        private static readonly Counter _globalExceptions = Metrics
-          .CreateCounter("global_exceptions", "Number of global exceptions.");
+        private static Counter _globalExceptions;
+        private static Gauge _httpRequestsInProgress;
+        private static Gauge _httpRequestsTotal;
+        private static Histogram _httpRequestsDuration;
+        private const string _timerKey = "PrometheusHttpRequestModule.Timer";
 
-        private static readonly Gauge _httpRequestsInProgress = Metrics
-            .CreateGauge("http_requests_in_progress", "The number of HTTP requests currently in progress");
+        /// <summary>
+        ///
+        /// </summary>
+        public PrometheusHttpRequestModule()
+        {
+            if (_globalExceptions == null)
+            {
+                _globalExceptions = Metrics
+                    .CreateCounter("global_exceptions", "Number of global exceptions.");
+            }
 
-        private static readonly Gauge _httpRequestsTotal = Metrics
-            .CreateGauge("http_requests_received_total", "Provides the count of HTTP requests that have been processed by this app",
-                new GaugeConfiguration { LabelNames = new[] { "code", "method", "controller", "action" } });
+            if (_httpRequestsInProgress == null)
+            {
+                _httpRequestsInProgress = Metrics
+                    .CreateGauge("http_requests_in_progress", "The number of HTTP requests currently in progress");
+            }
 
-        private static readonly Histogram _httpRequestsDuration = Metrics
-            .CreateHistogram("http_request_duration_seconds", "The duration of HTTP requests processed by this app.",
-                new HistogramConfiguration { LabelNames = new[] { "code", "method", "controller", "action" } });
+            if (_httpRequestsTotal == null)
+            {
+                _httpRequestsTotal = Metrics
+                    .CreateGauge("http_requests_received_total", "Provides the count of HTTP requests that have been processed by this app",
+                        new GaugeConfiguration {LabelNames = new[] {"code", "method", "controller", "action"}});
+            }
 
-        private const string _timerKey = "PrometheusHttpRequestModule.Timer";
+            if (_httpRequestsDuration == null)
+            {
+                _httpRequestsDuration = Metrics
+                    .CreateHistogram("http_request_duration_seconds", "The duration of HTTP requests processed by this app.",
+                        new HistogramConfiguration {LabelNames = new[] {"code", "method", "controller", "action"}});
+            }
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="bucektSize"></param>
+        public PrometheusHttpRequestModule(double[] bucektSize)
```

Comment:
If you change this constructor so that it looks like this:

```
public PrometheusHttpRequestModule(double[] bucektSize = null)
```

Does that mean you don't need the other constructor at all?

Reply:
(Also, please fix the spelling mistake :)